app/repositories/SessionRepository.scala

```diff
@@ -16,105 +16,59 @@
-  private val lastUpdatedIndex = Index.apply(BSONSerializationPack)(
-    key = Seq(lastUpdatedIndexKey -> IndexType.Ascending),
-    name = Some(lastUpdatedIndexName),
-    expireAfterSeconds = Some(cacheTtl),
-    options = BSONDocument.empty,
-    unique = false,
-    background = false,
-    dropDups = false,
-    sparse = false,
-    version = None,
-    partialFilter = None,
-    storageEngine = None,
-    weights = None,
-    defaultLanguage = None,
-    languageOverride = None,
-    textIndexVersion = None,
-    sphereIndexVersion = None,
-    bits = None,
-    min = None,
-    max = None,
-    bucketSize = None,
-    collation = None,
-    wildcardProjection = None
+@Singleton
+class DefaultSessionRepository @Inject()(val mongo: MongoComponent,
+                                          val config: FrontendAppConfig
+                                        )(implicit val ec: ExecutionContext) extends PlayMongoRepository[UserAnswers](
+    collectionName = "user-answers",
+    mongoComponent = mongo,
+    domainFormat = Format(UserAnswers.reads,UserAnswers.writes),
+    indexes = Seq(
+      IndexModel(
+        Indexes.ascending("lastUpdated"),
+        IndexOptions()
+          .name("user-answers-last-updated-index")
+          .expireAfter(config.cachettlInSeconds, TimeUnit.SECONDS).unique(false))
+    ), replaceIndexes = config.dropIndexes
   )
+  with SessionRepository
+{
 
-  def ensureTtlIndex(collection: JSONCollection): Future[Unit] = {
-    collection.indexesManager.ensure(lastUpdatedIndex) flatMap {
-      newlyCreated =>
-        // false if the index already exists
-        if (!newlyCreated) {
-          for {
-            _ <- collection.indexesManager.drop(lastUpdatedIndexName)
-            _ <- collection.indexesManager.ensure(lastUpdatedIndex)
-          } yield ()
-        } else {
-          Future.successful(())
-        }
-    }
-  }
+  override def get(id: String): Future[Option[UserAnswers]] = {
 
-  val started: Future[Unit] =
-    collection.flatMap(ensureTtlIndex).map(_ => ())
+    val selector = Filters.equal("_id", id)
+    collection.find(selector).headOption()
+  }
 
-  override def get(id: String): Future[Option[UserAnswers]] =
-    collection.flatMap(_.find(Json.obj("_id" -> id), None).one[UserAnswers])
 
   override def set(userAnswers: UserAnswers): Future[Boolean] = {
 
-    val selector = Json.obj(
-      "_id" -> userAnswers.id
-    )
-
-    val modifier = Json.obj(
-      "$set" -> (userAnswers copy (lastUpdated = LocalDateTime.now))
-    )
+    val selector = equal("_id" , userAnswers.id)
+    val newUser = userAnswers.copy(lastUpdated = LocalDateTime.now)
+    val replaceOptions = new ReplaceOptions().upsert(true)
 
-    collection.flatMap {
-      _.update(ordered = false)
-        .one(selector, modifier, upsert = true).map {
-          lastError =>
-            lastError.ok
-      }
+    collection.replaceOne(selector, newUser, replaceOptions) .headOption() .map(_.exists(_.wasAcknowledged()))
```

Comment:
could you remove the whitespace before each of the `.`
e.g. `replaceOptions) .headOption()` becomes `replaceOptions).headOption()`

(this is my fault I wrote the code slightly wrong on the last review)